docs/corruption-detection-and-recovery.md

```diff
@@ -0,0 +1,62 @@
+# etcd Corruption Detection and Recovery
+
+Currently, in etcd 3.5.0-3.5.2, there is a low probability risk of data corruption across the etcd members [Upstream Issue](https://github.com/etcd-io/etcd/issues/13766), [Openshift Documentation](https://access.redhat.com/solutions/6849521). There is a mitigation coming to the Openshift Cluster etcd Operator that enables the etcd 'experimental-initial-corrupt-check' which should help detect corruption and stop the bad member from joining the cluster.
+This document will outline some ways to verify if an etcd member is corrupted so that you can be more confident that the corruption recovery steps are necessary and should be taken.
+
+# Corruption Diagnosis
+Unfortunately, there's no single command that will give a precise 'corrupt/not corrupt' at this time; however, if most or all of the following return that there might be corruption, then it's likely that there is corruption and recovery steps should be taken. Please note, that a single failure in the following commands does not necessarily mean that corruption has occurred.
+
+## Check Prometheus for etcd revision divergence
+Revision divergence is a natural happening of etcd since it takes time to sync a revision to all members; however, continuing divergence is an indication of corruption. If you have access to run Prometheus queries, you can check the following to graph the current revisions to visualize any divergence.
+
+`max by (pod,job) (etcd_debugging_mvcc_current_revision)`
+
+---
+
+## Get dbSize and revision differences
+One indicator that there might be corruption is a 'significant' (greater than 10%) difference of etcd database size and revision number.
+
+### List the etcd pods
+`oc get pods -n openshift-etcd | grep -v etcd-quorum-guard | grep etcd`
+### Log into one of the nodes (choose a healthy node).
+`oc rsh -n openshift-etcd [etcd-pod]`
+
+### Show the db usage, these should be all the same or very similar
+`etcdctl endpoint status -w table --cluster`
+### Show revisions, these should be all the same or very similar
+`etcdctl endpoint status -w fields --cluster`
+
+---
+
+## Check leader elections
+Frequent and failing leader elections are a symptom of the etcd cluster having issues.
+
+### List the etcd pods
+`oc get pods -n openshift-etcd | grep -v etcd-quorum-guard | grep etcd`
+### Check the logs for any failures of leader elections
+`oc logs -n openshfit-etcd [etcd-pod] | grep "failed to renew lease"`
+
+---
+
+## Check the etcd pods for marshalling errors
+Marshalling errors, `failed to unmarshal mvccpb.KeyValue`, from the following command are an indication of corruption; however, no matches does not mean there isn't corruption.
+
+`oc -n openshift-etcd get -o yaml pods | grep mvccpb.KeyValue`
+
+---
+
+## Check the output of the startup corruption check
```

Comment:
Could a step at this point be to manually modify the static pod manifests on disk to add the check?  If it fails, then you have a problem, if it doesn't fail, then you're likely ok.